pkg/frontend/querymiddleware/split_by_instant_interval.go

```diff
@@ -28,6 +28,48 @@ type splitByInstantIntervalMiddleware struct {
 
 	splitEnabled  bool
 	splitInterval time.Duration
+
+	instantQuerySplittingMetrics
+}
+
+type instantQuerySplittingMetrics struct {
+	splittingAttempts    prometheus.Counter
+	mappedSplitQueries   *prometheus.CounterVec
+	splitQueries         prometheus.Counter
+	splitQueriesPerQuery prometheus.Histogram
+}
+
+// Parsing evaluation result used in instantQuerySplittingMetrics
+const (
+	SuccessKey = "success"
+	FailureKey = "failure"
+	NoopKey    = "noop"
+)
+
+func newInstantQuerySplittingMetrics(registerer prometheus.Registerer) instantQuerySplittingMetrics {
+	return instantQuerySplittingMetrics{
+		splittingAttempts: promauto.With(registerer).NewCounter(prometheus.CounterOpts{
+			Namespace: "cortex",
+			Name:      "frontend_instant_query_splitting_rewrites_attempted_total",
+			Help:      "Total number of instant queries the query-frontend attempted to split.",
+		}),
+		mappedSplitQueries: promauto.With(registerer).NewCounterVec(prometheus.CounterOpts{
+			Namespace: "cortex",
+			Name:      "frontend_instant_query_sharding_rewrites_succeeded_total",
```

Comment:
> frontend_instant_query_sharding_rewrites_succeeded_total

Change to "frontend_instant_query_splitting_rewrites_succeeded_total" to keep it consistent with "frontend_instant_query_splitting_rewrites_attempted_total".

Reply:
😮 good catch, completely missed that one 😅